### DISCUSSION TITLE
FastifyDeprecation: Request#context property access is deprecated

### DISCUSSION THREAD

**Discussion by AxelTahmid** [fastify/fastify#4315] [General] [answered]
Hello, just looking for the documentation to resolve this warning, 

```
[FSTDEP012] FastifyDeprecation: Request#context property access is deprecated. Please use "Request#routeConfig" or "Request#routeSchema" instead for accessing Route settings. The "Request#context" will be removed in `fastify@5`.
(Use `node --trace-warnings ...` to show where the warning was created) 
```

I am quite new to this, would appreciate any help regarding this. using v4.7 here.

**Comment by Eomm** [comment, marked as answer]
As it says the request.context is deprecated since this morning with the new fastify release.

We have introduced new getters on its behalf that are written in the warning message. 

We did not updated all the plugins accordingly, so if you can tell us which plugin triggers that error, we could update it 👍🏼

**Reply by AxelTahmid** [reply]
sorry for the delayed response, I am not entirely sure what caused this, but below are the plugins I am using. 

```
    "@fastify/cors": "^8.1.0",
    "@fastify/formbody": "^7.3.0",
    "@fastify/helmet": "^9.1.0",
    "@fastify/jwt": "^6.3.2",
    "@fastify/multipart": "^7.2.0",
    "@fastify/redis": "^6.0.0",
    "@fastify/sensible": "^5.1.1",
```

But I got the answer I needed, it being thrown by a plugin.

**Reply by Eomm** [reply]
Checked all the modules and only helmet had the issue that has been fixed:

https://github.com/fastify/fastify-helmet/releases/tag/v10.0.1

**Reply by radoh** [reply]
I also get this error (when I tried to access `request.context` in my preHandler hook), but when I try to use `request.routeConfig`, TypeScript complains that
```
TS2339: Property 'routeConfig' does not exist on type 'FastifyRequest ...>>'.
```
I am on fastify 4.7.0. Is this just a problem of missing type definitions somewhere?

**Reply by Eomm** [reply]
could you check `npm ls fastify`?

I think you should see if the plugin has its own version, we must bump it in the module

Meanwhile I will try this:
https://github.com/fastify/fastify-helmet/pull/205
